File: benchmarks/dx/claude_agent_sdk/errors/malformed_prompt_template.py
# SPDX-License-Identifier: AGPL-3.0-only
"""Bug 3: Malformed prompt template — unclosed {{variable}} placeholder.

Claude Agent SDK error quality analysis:
The SDK takes system prompts as raw strings. No template engine.
A malformed placeholder like {{name} (unclosed) is passed verbatim to
the LLM. The SDK performs no validation — the string is sent as-is.
  (No error raised. Behaviour: LLM sees the broken template literally.)
Score: file/line — 0 (no error raised)
       names problem — 0 (no feedback at all)
       suggests fix — 0 (developer must find the broken template manually)
Total diagnostic score: 0/30 → 0.0/10

Same result as Argentor: raw strings are not validated.
Neither framework uses a template engine, so both share this weakness.
Frameworks that DO catch this: LangChain (raises ValueError at construction).
"""
import anthropic

client = anthropic.Anthropic()

# BUG: unclosed template placeholder — {{name} missing closing brace
# The SDK sends this string verbatim to Claude. No error is raised.
malformed_system = "You are an assistant for {{name}. Answer their question."

response = client.messages.create(
    model="claude-sonnet-4-5",
    max_tokens=256,
    system=malformed_system,  # sent as-is — no validation
    messages=[{"role": "user", "content": "Hello"}],
)
print(response.content[0].text)

File: benchmarks/dx/argentor/errors/malformed_prompt_template.rs
//! Bug 3: Malformed prompt template — unclosed {{variable}} placeholder.
//!
//! Expected error quality:
//! Argentor validates prompt templates at compile time (type system) or
//! at agent construction. A raw string with {{unclosed passes through silently —
//! Argentor does NOT validate free-form system prompts (by design).
//! Result: the malformed template is sent as-is to the LLM. No error.
//! Score: file/line — 0 (no error raised)
//!        names problem — 0 (silent pass-through)
//!        suggests fix — 0 (no feedback)
//! Total diagnostic score: 0/30 → 0.0/10
//!
//! NOTE: This is a genuine weakness. Structured prompt libraries (e.g.,
//! LangChain's PromptTemplate) catch this at construction time. Argentor
//! takes raw strings and trusts the developer to validate them.

use argentor_agent::{AgentRunner, ModelConfig};

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    let config = ModelConfig {
        model: "claude-sonnet-4".to_string(),
        max_tokens: 256,
        ..Default::default()
    };
    let mut agent = AgentRunner::new(config)?;

    // BUG: unclosed template variable — {{name} missing closing brace
    // Argentor will send this string verbatim to the LLM without error.
    let malformed_system = "You are an assistant for {{name}. Answer their question.";
    let _r = agent.run(malformed_system, "Hello").await?;
    Ok(())
}
